azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -507,13 +508,14 @@ private boolean isDeployToSlot() {
     }
 
     private ApplicationInsightsComponent getOrCreateApplicationInsights(boolean enableCreation) throws AzureAuthFailureException, AzureExecutionException {
-        final AzureCredentialWrapper credentials = getAzureCredentialWrapper();
-        if (credentials == null) {
+        final Account account = com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).account();
+        if (!account.isAuthenticated()) {
```

Comment:
we can simplify much of our code if we throw exception inside `account()` instead of check everywhere